src/components/RequestsTable/FilterRequests.tsx

```diff
@@ -0,0 +1,452 @@
+                      {grade}
+                    </Option>
+                  ))}
+                  {GENERALGRADES.map((grade) => (
+                    <Option key={grade} value={grade}>
+                      {grade}
+                    </Option>
+                  ))}
+                </Combobox>
+              )}
+            />
+          </Field>
+          <hr />
+          <Field label="Office Symbol">
+            <Controller
+              name="officeSymbol"
+              control={control}
+              render={({ field }) => <Input type="search" {...field} />}
+            />
+          </Field>
+          <hr />
+          <Field label="Created By">
+            <Controller
+              name="Author"
+              control={control}
+              render={({ field }) => (
+                <PeoplePicker
+                  ariaLabel="Requestor"
+                  selectedItems={field.value ?? []}
+                  updatePeople={async (items) => {
+                    if (items?.[0]?.Title) {
+                      const userId = (
+                        await spWebContext.web.ensureUser(items?.[0].EMail)
+                      ).data.Id;
+                      field.onChange({ ...items[0], Id: userId });
+                    } else {
+                      field.onChange([]);
+                    }
+                  }}
+                />
+              )}
+            />
+          </Field>
+          <hr />
+          <Field label="Stage">
+            <Controller
+              name="stage"
+              control={control}
+              render={({ field }) => (
+                <Combobox
+                  clearable
+                  autoComplete="on"
+                  {...field}
+                  selectedOptions={[field.value ?? ""]}
+                  onOptionSelect={(_event, data) => {
+                    field.onChange(data.optionValue ?? "");
+                  }}
+                  value={field.value}
+                >
+                  {STAGES.map((stage) => (
+                    <Option key={stage.key} text={stage.text} value={stage.key}>
+                      {stage.text}
+                    </Option>
+                  ))}
+                </Combobox>
+              )}
+            />
+          </Field>
+          <hr />
+          <Field label="Created After">
+            <Controller
+              name="afterDate"
+              control={control}
+              render={({ field }) => (
+                <>
+                  <DatePicker
+                    formatDate={(date?: Date) => {
+                      return date?.toLocaleDateString() ?? "";
+                    }}
+                    onSelectDate={field.onChange}
+                    {...field}
+                  />
+                  <Button
+                    appearance="primary"
+                    style={{ marginTop: "2px" }}
+                    onClick={() => field.onChange(undefined)}
```

Comment:
You should make the dates null when they aren't defined, otherwise the DatePicker goes between being uncontrolled and being controlled, which causes issues.  When it is undefined, and I set a filter, apply it, then open the filter drawer again, I can't clear the filter.